service/controller/resource/cleanuprecordsets/delete.go

```diff
@@ -0,0 +1,115 @@
+package cleanuprecordsets
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var hostedZones []*route53.HostedZone
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding hosted zones for tenant cluster %#q", key.ClusterID(&cr)))
+
+		hostedZonesInput := &route53.ListHostedZonesByNameInput{}
+
+		o, err := cc.Client.TenantCluster.AWS.Route53.ListHostedZonesByName(hostedZonesInput)
```

Comment:
IIRC there is some paging. But I may be wrong.

Reply:
Will check this

Reply:
```
Route 53 returns up to 100 items in each response
```

It should be fine with this. We don't have hundreds of clusters